data_structures/queue/monotonic_queue.py

```diff
@@ -0,0 +1,39 @@
+from __future__ import annotations
+
+from .double_ended_queue import Deque
+
+arr = [1, 3, -1, -3, 5, 3, 6, 7]
+window_size = 3
+expect = [3, 3, 5, 5, 6, 7]
+
+
+def max_sliding_window(arr: list[float], window_size: int) -> list[float]:
```

Comment:
Can we take the same approach with this algorithm?
https://github.com/TheAlgorithms/Python/pull/10273#discussion_r1355794522

Seems appropriate for a sliding_window algorithm.

Reply:
It is a bit different because the queue has to be monotonically decreasing while calculating the max value in a window.

Reply:
If it does not make sense then we can close the pull request.

Reply:
Hi, I resolved the issue of using collections.deque instead of double_ended_queue. 

I feel the max sliding window is still an important algorithm which is a good use case of the queue (it is similar to next_greater_element in stack).

Reply:
@tianyizheng02 Is my iterator, not list request above unreasonable in this algorithm?